wasmcloud/tests/smtp_integration_test.rs

```diff
@@ -0,0 +1,457 @@
+        .with_wait_for(WaitFor::message_on_either_std("connected"))
+        .with_env_var("WADM_NATS_SERVER", bridge_ip.to_string())
+        .with_network("bridge")
+        .with_container_name("wadm_integration_test")
+        // .with_log_consumer(show_logs)
+        .start()
+        .await
+        .expect("Failed starting wadm")
+}
+
+async fn start_mail_server() -> ContainerDef {
+    // NOTE: Using this mail server as it implements authentication
+    GenericImage::new("rnwood/smtp4dev", "latest")
+        .with_exposed_port(MAILCATCHER_SMTP_PORT.tcp())
+        .with_exposed_port(MAILCATCHER_API_PORT.tcp())
+        .with_wait_for(WaitFor::message_on_either_std("Now listening on"))
+        //.with_log_consumer(show_logs)
+        .with_network("bridge")
+        .with_container_name("mail_integration_test")
+        .start()
+        .await
+        .expect("Failed to start catcher")
+}
+
+async fn start_everything() -> (ContainerDef, ContainerDef, ContainerDef, ContainerDef) {
+    let nats = start_nats().await;
+
+    let bridge_ip = nats.get_bridge_ip_address().await.unwrap();
+    let wasmcloud = start_wasmcloud(&bridge_ip).await;
+    let wadm = start_wadm(&bridge_ip).await;
+    let catcher = start_mail_server().await;
+
+    (nats, wasmcloud, wadm, catcher)
+}
+
+async fn publish_components(bridge_ip: &IpAddr) {
+    let client_connect = ClientConnectOptions {
+        url: Some(format!("{}:{}", bridge_ip, NATS_PORT)),
+        ..Default::default()
+    };
+
+    let client = wadm_client::Client::new("default", None, client_connect)
+        .await
+        .expect("Failed client");
+
+    let manifest = "wadm.test.yaml"
+        .load_manifest()
+        .await
+        .expect("Failed to load manifest");
+
+    if let Ok(_existing_manifest) = client.get_manifest(&manifest.metadata.name, None).await {
+        return;
+    }
+
+    client
+        .put_and_deploy_manifest(&manifest)
+        .await
+        .expect("Failed to put and deploy manifest");
+    let mut stream = client
+        .subscribe_to_status(&manifest.metadata.name)
+        .await
+        .expect("Failed to subscribe");
+    loop {
+        let message = stream.next().await.expect("Failed to read next message");
+        let payload: serde_json::Value =
+            serde_json::from_slice(&message.payload).expect("Failed to parse payload");
+        if payload["status"]["type"] == "deployed" {
+            // NOTE: Wait for x-seconds to wait so that the http-interface is online
+            sleep(Duration::new(2, 0)).await;
+            break;
+        }
+    }
+}
+
+async fn get_mail_text(api_port: u16) -> String {
+    let messages_response = reqwest::get(format!("http://127.0.0.1:{}/api/messages", api_port))
+        .await
+        .expect("Failed to get");
+
+    let _json: serde_json::Value = messages_response.json().await.unwrap();
+    let message_id = _json["results"][0]["id"].as_str().unwrap();
```

Comment:
maybe refactor to have the calling of the mailcatcher api a bit nicer (you now copy pasted it)